src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,343 @@
+  // Putting `aria-hidden` on it as it won't contain any readable information
+  this.$indicatorContainer = document.createElement('div')
+  this.$indicatorContainer.className = 'govuk-exit-this-page__indicator'
+  this.$indicatorContainer.setAttribute('aria-hidden', 'true')
+
+  // Create three 'lights' and place them within the container
+  for (var i = 0; i < 3; i++) {
+    var $indicator = document.createElement('div')
+    $indicator.className = 'govuk-exit-this-page__indicator-light'
+    this.$indicatorContainer.appendChild($indicator)
+  }
+
+  // Append it all to the module
+  this.$button.appendChild(this.$indicatorContainer)
+}
+
+/**
+ * Update whether the lights are visible and which ones are lit up depending on
+ * the value of `keypressCounter`.
+ */
+ExitThisPage.prototype.updateIndicator = function () {
+  // Show or hide the indicator container depending on keypressCounter value
+  if (this.keypressCounter > 0) {
+    this.$indicatorContainer.classList.add('govuk-exit-this-page__indicator--visible')
+  } else {
+    this.$indicatorContainer.classList.remove('govuk-exit-this-page__indicator--visible')
+  }
+
+  // Turn out all the lights
+  var $lightsOn = this.$indicatorContainer.querySelectorAll('.govuk-exit-this-page__indicator-light--on')
+  nodeListForEach($lightsOn, function ($light) {
+    $light.classList.remove('govuk-exit-this-page__indicator-light--on')
+  })
+
+  // Turn on the ones we want on
+  var $lightsQueried = this.$indicatorContainer.querySelectorAll('.govuk-exit-this-page__indicator-light')
+  nodeListForEach($lightsQueried, function ($light, index) {
+    if (index < this.keypressCounter) {
+      $light.classList.add('govuk-exit-this-page__indicator-light--on')
+    }
+  }.bind(this))
+}
+
+/**
+ * Initiates the redirection away from the current page.
+ * Includes the loading overlay functionality, which covers the current page with a
+ * white overlay so that the contents are not visible during the loading
+ * process. This is particularly important on slow network connections.
+ *
+ * @param {MouseEvent} [event] - mouse click event
+ */
+ExitThisPage.prototype.exitPage = function (event) {
+  var redirectUrl = this.$button.getAttribute('href')
+
+  if (typeof event !== 'undefined' && event.target instanceof HTMLElement) {
+    event.preventDefault()
+    redirectUrl = event.target.getAttribute('href')
+  }
+
+  this.$updateSpan.innerText = ''
+
+  // Blank the page
+  // As well as creating an overlay with text, we also set the body to hidden
+  // to prevent screen reader and sequential navigation users potentially
+  // navigating through the page behind the overlay during loading
+  document.body.classList.add('govuk-exit-this-page-hide-content')
+  this.$overlay = document.createElement('div')
+  this.$overlay.className = 'govuk-exit-this-page-overlay'
+  this.$overlay.setAttribute('role', 'alert')
+
+  document.body.appendChild(this.$overlay)
+  this.$overlay.innerText = this.i18n.t('activated')
```

Comment:
Not 100% sure but might be able to avoid an extra paint by doing these the other way around?If it needs to be this way round so that it's announced it might be worth adding a comment.

Reply:
I think I flagged this at an earlier point and it was the case that the element and text need to be inserted separately for screen readers to reliably pick it up.

Reply:
I'll add a comment with Beep's point